### PR TITLE
New method for  calcule spent time in week passed 

### PR DESCRIPTION
**Pull request by lamrani002**
# NEW|New method in project class by get total of element_duration and check if had a public holiday

### CODE REVIEW

**Comment by eldy** [comment]
Errors reported by pre-commit. See on "details".

File: htdocs/projet/class/project.class.php [hunk 1]
```diff
@@ -2549,4 +2549,66 @@ public function getChildren()
...
+
+	/**
+	 * method for calcule weekly time spent and get a repport
+	 * @return string|int
+	 */
+	public function createWeeklyReport()
+	{
+		global $mysoc;
+		$lastWeekStart = new DateTime();
+		$lastWeekStart->modify('last monday -7 days');
```
Comment (eldy) [inline htdocs/projet/class/project.class.php:2561]: first day of week may vary depending on user setup.
Can you check if you can replace with dol_get_first_day_week() in date.lib.php

**Comment by lamrani002** [comment]
the email is still not inserted in the actioncomm I haven't processed it yet

File: htdocs/projet/class/project.class.php [hunk 2]
```diff
@@ -2549,4 +2549,86 @@ public function getChildren()
...
+			}
+			$reportContent .= '</table>';
+
+			require_once DOL_DOCUMENT_ROOT.'/core/class/CMailFile.class.php';
+			$subject = 'Rapport hebdomadaire des temps travaillés';
+
+			$from = getDolGlobalString('MAIN_MAIL_EMAIL_FROM');
+
+
+			$mail = New CMailFile($subject, $to, $from, $reportContent);
```
Comment (eldy) [inline htdocs/projet/class/project.class.php:2626]: new in lowercase

**Comment by lamrani002** [comment]
conten

**Pull request merged**